dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataanalysis/FollowupAnalysisParams.java

```diff
@@ -0,0 +1,116 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dataanalysis;
+
+import java.util.Date;
+import java.util.List;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.dataset.DataSet;
+import org.hisp.dhis.period.Period;
+
+/**
+ * Request query parameters for followup analysis.
+ *
+ * Mandatory are:
+ * <ul>
+ * <li>at least one {@link #ouParent}</li>
+ * <li>either {@link #de} or {@link #ds}</li>
+ * <li>either {@link #pe} or {@link #startDate} and {@link #endDate}</li>
+ * </ul>
+ *
+ * Optional are:
+ * <ul>
+ * <li>{@link #coc} (derived from {@link #de} if empty)</li>
+ * <li>{@link #page} (default 1)</li>
+ * <li>{@link #pageSize} (default 50)</li>
+ * </ul>
+ *
+ * @author Jan Bernitt
+ */
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor( access = AccessLevel.PUBLIC )
+@AllArgsConstructor( access = AccessLevel.PRIVATE )
+public final class FollowupAnalysisParams
+{
+    /**
+     * {@link org.hisp.dhis.dataelement.DataElement} UIDs
+     *
+     * Only considered when {@link #ds} is non-empty. Otherwise the elements
+     * considered is based on {@link DataSet#getDataElements()}.
+     */
+    private List<String> de;
+
+    /**
+     * {@link org.hisp.dhis.dataset.DataSet} UIDs
+     *
+     * If specified it as the same effect as if {@link #de} had been filled with
+     * the union of all {@link DataSet#getDataElements()}.
+     */
+    private List<String> ds;
+
+    /**
+     * {@link org.hisp.dhis.category.CategoryOptionCombo} UIDs
+     *
+     * If not specified (empty) it has the same effect as if it holds the union
+     * of all {@link DataElement#getCategoryOptionCombos()} that are
+     * {@link ValueType#isNumeric()}.
+     */
+    private List<String> coc;
+
+    /**
+     * {@link org.hisp.dhis.organisationunit.OrganisationUnit} UIDs of the
+     * parent(s) whose children should be included
+     */
+    private List<String> ouParent;
+
+    /**
+     * As an alternative to providing {@link #startDate} and {@link #endDate} a
+     * {@link Period} ISO string can be provided which is just used to derive
+     * the start and end date from it.
+     */
+    private String pe;
+
+    private Date startDate;
+
+    private Date endDate;
+
+    private int page;
```

Comment:
I think we can borrow from `OutlierDetectionRequest` and have `int maxResults` instead. The reason is that we don't sort/order by anything in the HQL query, and so paging becomes undefined. Paging is also tricky with data values in general. Instead users can filter by org unit, by time period, by data element etc to narrow their scope.